docs/docsite/rst/presentations/ansible-testing.rst

```diff
@@ -0,0 +1,438 @@
+     * Important: Reduce regression, help support Python 3
+
+   * Tailor content:
+
+     * Q: Show of hands for version, devel, 2.3, -> 1.9? - upgrade!
+     * Q: Who's raised a PR, should be getting easier on each release, especially modules
+
+   * ReStructured Text: Slides will be made available online, which include speakers notes
+   * All docs & module docs can be edited in GitHub UI, no knowledge of Git needed!
+   * Questions welcome at any point - Will ask Mark to make a note
+
+
+Requirements
+============
+
+What do we want from testing Ansible:
+
+* Prevent regressions & speed up development
+* Increase code quality
+* Better readable code, improve collaboration
+
+* Ability to run same tests locally as run by CI (Shippable)
+* Different OS, Python versions (2.4+, 3.5+)
+* Reasonable run time
+
+.. container:: handout
+
+   * Local run - common request from the community
+   * Increasing the matrix of Python & OS versions vs timely feedback
+   * Main aim: Stable product, find issues sooner, speed up PR work flow
+
+Solution: ansible-test
+======================
+
+* Shippable
+* Different OS & Python: Docker & tox
+* Local runs: Shippable is just a wrapper
+* Duration: Inspect git diff. File to test mapping.
+* Nightly runs with code coverage analysis.
+
+
+.. container:: handout
+
+   * Intro Matt Clay
+
+     * Principal Engineer in USA
+     * Full time on Testing, great Python programmer
+
+   * Q: If I said Shippable or Travis, who would know what they are:
+
+     * Shippable is a Continuous Integration (CI) tool, like Travis
+     * Integrate with GitHub to test (PRs) + merges
+     * Originally used Travis, moved as Shippable is cheaper
+     * 52 Concurrent jobs as of May 2017
+
+   * Local Runs
+
+     * Different OS + Local runs fit together
+     * Shippable is just a wrapper, all logic in ansible-test
+     * tox: Sanity & Unit tests
+
+   * Duration
+
+     * Tracking of Python imports
+     * No need to run integration tests on RST changes
+
+   * Code coverage is ~1.5 - 3x as slow (integration - Unit tests)
+
+   * Cost vs benefits
+
+     * Interrupts the workflow (wait until CI is finished, change, amend, push)
+     * Not everything can be tested locally
+     * Cosmetic failures can be frustrating
+     * Do the benefits outweigh the costs ? We sure hope so :-)
+
+
+Types of tests
+==============
+
+* **compile**
+
+  * Test python code against a variety of Python versions.
+
+* **sanity**
+
+  * Sanity tests are made up of scripts and tools used to perform static code analysis.
+  * The primary purpose of these tests is to enforce Ansible coding standards and requirements.
+
+* **integration**
+
+  * Functional tests of modules and Ansible core functionality.
+  * Best for testing modules and core features (become, ``with_*``, add_host)
+
+* **units**
+
+  * Tests directly against individual parts of the code base.
+  * Best for testing libraries (``module_utils``)
+
```

Comment:
This section with bullets: remain consistent - add periods onto each statement or remove them.